gotchas/gotcha-2.md

```diff
@@ -0,0 +1,91 @@
+## Haskell.nix Dependency Build Problems
+
+Sometimes the dependencies of a project fail to build and you will
+need to add overrides to the Haskell.nix config to fix it.
+
+This file lists build problems that you may encounter and their solutions.
+
+### `No attribute: hsc2hs`
+
+This is a bug in Haskell.nix which you will see if you have a
+dependency which uses `hsc2hs` as a build tool.
+
+Work around it by adding `hsc2hs` to the `extra-deps` of `stack.yaml`.
+
+### Errors about `Divisible` when compiling `contravariant`
+
+```
+[1 of 3] Compiling Data.Functor.Contravariant.Divisible ( src/Data/Functor/Contravariant/Divisible.hs, /home/refold/code/scrive/contravariant-1.5/dist-newstyle/build/x86_64-linux/ghc-8.6.4/contravariant-1.5/build/Data/Functor/Contravariant/Divisible.o ) [Data.StateVar changed]
+
+src/Data/Functor/Contravariant/Divisible.hs:233:10: error:
+    • Could not deduce (Contravariant (Backwards f))
+        arising from the superclasses of an instance declaration
+      from the context: Divisible f
+        bound by the instance declaration
+        at src/Data/Functor/Contravariant/Divisible.hs:233:10-47
+    • In the instance declaration for ‘Divisible (Backwards f)’
+```
+
+This is a bug in GHC 8.6.1 through to 8.6.4. It shipped with the wrong
+version of the transformers package .
+
+Work around this by adding the following to `stack.yaml`:
+
+```yaml
+extra-deps:
+- process-1.6.5.0
+- transformers-0.5.6.2
+```
+
+### Errors about package version bounds
+
+This generally won't happen if you are using the Cabal solver
+(`plan-to-nix`) or a Stackage LTS.
+
+But if you need to "jailbreak" a package, add a module to your package
+set like this:
+
+```nix
+{
+  packages.transformers.components.library.doExactConfig = true;
+}
+```
+
+### `attribute ghc864 missing`
+
+This happens when a compiler is not available in the Nixpkgs package
+collection that you are using. Nixpkgs will only keep one compiler of
+each major version.
+
+You may need to use a different Nixpkgs channel.
+
+If using `stack-to-nix`, you could choose a different Stackage snapshot.
+
+If using `plan-to-nix`, you could regenerate your plan and nix with a
+compiler version that exists in your Nixpkgs.
+
```

Comment:
If you use `iohk-nix`/`input-output-hk/nixpkgs`, bumping that might help as well.